numpy/core/multiarray.py

```diff
@@ -12,10 +12,9 @@
 from ._multiarray_umath import *  # noqa: F403
 # These imports are needed for backward compatibility,
 # do not change them. issue gh-15518
-# _get_ndarray_c_version is semi-public, on purpose not added to __all__
 from ._multiarray_umath import (
     _fastCopyAndTranspose, _flagdict, _insert, _reconstruct, _vec_string,
-    _ARRAY_API, _monotonicity, _get_ndarray_c_version, _set_madvise_hugepage,
```

Comment:
Not sure we should remove the first, if it is has a "semi-public" comment.  The second could be removed, but then you need to change the one place where it is used as well (in `__init__.py`).